som/display.go

```diff
@@ -0,0 +1,120 @@
+package som
+
+import (
+	"encoding/xml"
+	"fmt"
+	"io"
+	"math"
+
+	"github.com/gonum/matrix/mat64"
+)
+
+// Creates an SVG representation of the U-Matrix of the given codebook.
+// codebook is the codebook we're displaying the U-Matrix for,
+// coordsDims are the dimensions of the grid,
+// uShape is the shape of the grid,
+// title is the title of the output SVG,
+// and writer is the io.Writter to write the output SVG to.
+func UMatrixSVG(codebook *mat64.Dense, coordsDims []int, uShape string, title string, writer io.Writer) error {
+	xmlEncoder := xml.NewEncoder(writer)
+	// array to hold the xml elements
+	elems := []interface{}{h1{Title: title}}
+
+	rows, _ := codebook.Dims()
+	distMat, _ := DistanceMx("euclidean", codebook)
+	distMatRows, distMatCols := distMat.Dims()
+	coords, _ := GridCoords(uShape, coordsDims)
+	coordsDistMat, _ := DistanceMx("euclidean", coords)
+
+	// get maximum distance between codebook vectors
+	// maximum distance will be rgb(0,0,0)
+	MAX := 0.0
+	for i := 0; i < distMatRows; i++ {
```

Comment:
Can we replace this for loop with [Max()](https://godoc.org/github.com/gonum/matrix/mat64#Max) function? i.e.

`max := mat64.Max(distMat)`